db/migrate/20210923070806_add_star_rating_to_space.rb

```diff
@@ -0,0 +1,5 @@
+class AddStarRatingToSpace < ActiveRecord::Migration[6.1]
+  def change
+    add_column :spaces, :star_rating, :integer, null: false, default: 0
```

Comment:
Hm, what if we don't have any ratings yet? 

In the sketches this is shown like this:
![image](https://user-images.githubusercontent.com/14905290/134504991-2344414d-d66f-45da-b202-b86bcc242272.png)

How is this state represented, if not as `null`? 

Or should it be `0` if we have no ratings? If so, I guess the validator should allow for a 0 star rating too?

Reply:
I think it makes the most sense to represent 
![image](https://user-images.githubusercontent.com/2104029/134507259-1bd83d63-2d01-437c-bb2e-cf3eb472fb5a.png)
as 0

Reply:
I feel like `0` is less clear about what's actually going on than `nil` is. 

Willing to be convinced otherwise!

My thinking goes like this: 

If there's a number there, someone who doesn't know the whole code base will assume it's an aggregated score. But  the number `0` would not be an aggregated score. Instead, it's an indication that there is no aggregation. 

`nil` on the other hand makes it clear for anyone who encounters the value without context that there is no value stored here, and that there has been no aggregation (which is closer to the truth). `0` might instead lead them to believe that we could have values like 0.2 or 0.5, and that this space just has really bad reviews. 

Similarily, other places in the code base we would have to remember to check like this:

```ruby
show_rating if rating != 0
```
as ruby won't know that 0 is false/non-existent. 

As seen in irb here:
![image](https://user-images.githubusercontent.com/14905290/134509592-01e3d6f9-d247-4402-b9c4-f018b47a80ec.png)

With `nil`, ruby knows it's false, and we can instead do something like:
 
```ruby
show_rating if rating
```

To me, `nil` seems more semantically correct. 

What do you think?

Reply:
Yeah my main issue (which may not even come up) is that you cannot do math operations on nil like `5 * nil` this will throw an error. 
So we need to keep remembering that this value can be nil. Which probably isn't really an issue in this case but it's the reason why i am hesitant to use nil to represent something

Reply:
I think if we do math on the `0` then we have screwed up something 😂 , and should probably be told that it's not possilbe. 

If we ever average 0 with 5 and get 2.5, then we have a bug - as the 0 should never be used for aggregation. It's not a score, but a state. So to me, the math argument makes me even more certain that `nil` is correct for this use case! At least then we'll get an error if we try to do math with it.

Reply:
Okay, let's go with nil for this.